### ISSUE TITLE
Problem with MotionPlus

### ISSUE THREAD

**Issue by carlotes247** [Flafla2/Unity-Wiimote#1]
Hello Flafla2! What an incredible work you have done here! I'm trying to use the WiiMotion Plus with your code but it seems to not be recognizing the motionPlus. I am following the code example on your mainpage.dox, where you are suggesting on doing something like this:

``` C#
Wiimote remote; // Initialized previously
bool activated = false;

void Start() {
    remote.RequestIdentifyMotionPlus();
}

void Update() {
    if (!activated && remote.wmp_attached) {
        remote.ActivateWiiMotionPlus();
        activated = true;
    }

    if(activated && remote.current_ext == ExtensionController.MOTIONPLUS) {
        MotionPlusData data = remote.MotionPlus; // data!
        float dPitch = data.PitchSpeed;
        // Use the data...
    }
}
```

And I wrote into WiiMoteDemo.cs a function:

``` C#
void RotateObjectAxis()
    {
        if (wiimote == null) return;
        Debug.Log("Entering RotateObject Method. WiiMotionPlusActive value: " + WiiMotionPlusActive.ToString());
        Debug.Log("wmp_attached value: " + wiimote.wmp_attached);

        //Code for the WiiMotionPlus
        if (!WiiMotionPlusActive && wiimote.wmp_attached)
        {
            Debug.Log("Activating MotionPlus...");
            wiimote.ActivateWiiMotionPlus();
            WiiMotionPlusActive = true;
        }
        Debug.Log(wiimote.current_ext.ToString());

        if (WiiMotionPlusActive && wiimote.current_ext == ExtensionController.MOTIONPLUS )
        {
            Debug.Log("Using MotionPlusData...");
            MotionPlusData motionPlusData = wiimote.MotionPlus; //Data!
            float dPitch = motionPlusData.PitchSpeed;
            float dYaw = motionPlusData.YawSpeed;
            float dRoll = motionPlusData.RollSpeed;

            BoxToMove.transform.rotation = new Quaternion(dPitch, dYaw, dRoll, 0f);
            Debug.Log("Pitch: " + dPitch.ToString() + " Yaw: " + dYaw.ToString() + " Roll: " + dRoll.ToString());
        }

    }
```

And another one:

``` C#
//Function to check motionPlus
    void CheckMotionPlus()
    {
        wiimote.RequestIdentifyWiiMotionPlus();
        Debug.Log("MotionPlusChecked!");
    }
```

Then, I define a button OnGUI that every time is pressed calls CheckMotionPlus(). Also, RotateObjectAxis() is being called on the Update loop right after the ReadWiimoteData(), as you can see here:

``` C#
void Update () {
        if (!WiimoteManager.HasWiimote()) { return; }

        wiimote = WiimoteManager.Wiimotes[0];

        int ret;
        do
        {
            ret = wiimote.ReadWiimoteData();
        } while (ret > 0);

        RotateObjectAxis();
}
```

Unfortunately, the "wmp_attached value" is always NONE when I'm having my motionPlus connected to the wiimote. If I connect a NUNCHUCK through the motionPlus, it only says NUNCHUCK, not MOTIONPLUS_NUNCHUCK. 

Any clue on how to make the MotionPlus work or is it maybe that I am using your code the wrong way?

Thanks a lot for the work and looking forward to hearing from you! I'm honestly impressed by hat you did here!

**Comment by Flafla2** [comment]
Hi!  Sorry I took so long - I didn't notice your post until now.  I have to admit that the WMP functionality has not been thoroughly tested.  I will look into it today.

However, I can say that the behavior regarding NUNCHUCK vs MOTIONPLUS_NUNCHUCK is by design.  MOTIONPLUS_NUNCHUCK refers to Nunchuck passthrough mode, which is currently not supported by WiimoteApi.  This is when the Wiimote reports data from both the nunchuck and the Wii Motion plus at the same time.

**Comment by Flafla2** [comment]
Hi again.  Upon further investigation I can confirm that this is a bug with RequestIdentifyWiiMotionPlus() and ActivateWiiMotionPlus().  I am working on a fix now and it should be up by the end of the day.

**Comment by Flafla2** [comment]
Fixed with e35618419d1375293e9d6a784eb08bddcd2b2762

**Comment by carlotes247** [comment]
Hello! Thank you very much for the answer and the fix! It's great!

I'll be working with it and testing it in the upcoming days, so if I find any other issue I'll write it down.

And once again, you should feel proud of what you coded here! :D
